central/metrics/custom/tracker/tracker_base.go

```diff
@@ -221,20 +228,44 @@ func (tracker *TrackerBase[Finding]) track(ctx context.Context, registry metrics
 
 // Gather the data not more often then maxAge.
 func (tracker *TrackerBase[Finding]) Gather(ctx context.Context) {
+	id, err := authn.IdentityFromContext(ctx)
+	if err != nil {
+		return
+	}
 	cfg := tracker.GetConfiguration()
-	gatherer := tracker.gatherer
+	gatherer := tracker.getGatherer(id.UID(), cfg)
 
 	// Return if is still running.
 	if !gatherer.running.CompareAndSwap(false, true) {
 		return
 	}
 	defer gatherer.running.Store(false)
 
-	if cfg == nil || cfg.period == 0 || time.Since(gatherer.lastGather) < cfg.period {
+	if cfg.period == 0 || time.Since(gatherer.lastGather) < cfg.period {
```

Comment:
Why did you remove the nil check here?

Reply:
Oh, I believe in some state of the code I had GetConfiguration never return nil. But I'll bring back the check, but hight in the code, because it may panic in `getGatherer()` already.